nemo_curator/modifiers/llm_pii_modifier.py

```diff
@@ -0,0 +1,195 @@
+
+        Args:
+            base_url (str): The base URL for the user's NIM
+            api_key (Optional[str]): The API key for the user's NIM, if needed.
+                Default is None.
+            model (str): The model to use for the LLM.
+                Default is "meta/llama-3.1-70b-instruct".
+            system_prompt (Optional[str]): The system prompt to feed into the LLM.
+                If None, a default system prompt is used.
+                Default prompt has been fine-tuned for "meta/llama-3.1-70b-instruct".
+            pii_labels (Optional[List[str]]): The PII labels to identify and remove from the text.
+                See documentation for full list of PII labels.
+                Default is None, which means all PII labels will be used.
+            language (str): The language to use for the LLM.
+                Default is "en" for English. If non-English, it is recommended
+                to provide a custom system prompt.
+
+        """
+        super().__init__()
+
+        self.base_url = base_url
+        self.api_key = api_key
+        self.model = model
+
+        if system_prompt is not None and pii_labels is not None:
+            warnings.warn(
+                "Custom system_prompt and custom pii_labels were both provided, "
+                "but the PII labels should already be included in the system prompt. "
+                "The pii_labels will be ignored."
+            )
+
+        if pii_labels is None:
+            pii_labels = PII_LABELS
+        if system_prompt is None:
+            self.system_prompt = get_system_prompt(pii_labels)
+        else:
+            self.system_prompt = system_prompt
+
+        if language != "en" and system_prompt is None:
+            warnings.warn(
+                "The default system prompt is only available for English text. "
+                "For other languages, please provide a custom system prompt. "
+                "Please refer to the default system prompt as a guide: "
+                "\n"
+                f"{get_system_prompt(pii_labels)}"
+                "\n"
+                "In particular, please ensure that the JSON schema is included in the system prompt exactly as shown: "
+                "\n"
+                f"{str(JSON_SCHEMA)}"
+            )
+        if language == "en" and system_prompt is not None:
+            warnings.warn(
+                "Using the default system prompt is strongly recommended for English text. "
+                "If you are customizing the system prompt, please refer to the default system prompt as a guide: "
+                f"{get_system_prompt(pii_labels)}"
+                "\n"
+                "In particular, please ensure that the JSON schema is included in the system prompt exactly as shown: "
+                "\n"
+                f"{str(JSON_SCHEMA)}"
+            )
+
+    def modify_document(self, text: str):
+        inferer = load_object_on_worker("inferer", self.load_inferer, {})
```

Comment:
Same here. 
```suggestion
        ```suggestion
        self._inferer_key = f"inferer_{id(self)}"
        inferer = load_object_on_worker(self._inferer_key, self.load_inferer, {})
```
```